equistore/operations/dot.py

```diff
@@ -0,0 +1,71 @@
+import numpy as np
+
+from equistore import TensorBlock, TensorMap
+
+from . import _dispatch
+
+
+def _dot_block(block1: TensorBlock, block2: TensorBlock) -> TensorBlock:
+    """dot product (row) among two `TensorBlocks`.
+    The `TensorBlocks` should have the same `properties`
+    """
+
+    assert np.all(block1.properties == block2.properties)
```

Comment:
I think this should be an exception with a better error message, since the user is likely to encounter it. I tend to prefer using assertions for internal consistency checks